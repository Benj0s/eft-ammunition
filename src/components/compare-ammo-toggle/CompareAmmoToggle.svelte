<script>
  import { get } from "svelte/store";
  import { compareAllAmmo } from "../../store.js";
  import Switch from "@smui/switch";
  import FormField from "@smui/form-field";

  let checked = get(compareAllAmmo);

  $: compareAllAmmo.update(() => checked);
</script>

<style>
  .compare-ammo-toggle {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin: 0 20px;
  }

  .ammo-toggle-label {
    color: hsla(0, 0%, 87.1%, 0.87);
  }
</style>

<div class="compare-ammo-toggle">
  <FormField>
    <Switch bind:checked />
    <span class="ammo-toggle-label" slot="label">Compare</span>
  </FormField>
</div>
