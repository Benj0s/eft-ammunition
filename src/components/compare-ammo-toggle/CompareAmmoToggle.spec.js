import CompareAmmoToggle from './CompareAmmoToggle.svelte'
import { render } from '@testing-library/svelte'

describe('CompareAmmoToggle', () => {
  it('should render the component with toggle off by default', async () => {
    const { getByRole, container } = render(CompareAmmoToggle)
    const switchEl = getByRole('switch')

    expect(switchEl.getAttribute('aria-checked')).toEqual('false')

    expect(container).toMatchSnapshot();
  })
})
