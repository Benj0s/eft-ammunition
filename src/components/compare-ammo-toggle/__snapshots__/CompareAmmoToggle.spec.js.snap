// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareAmmoToggle should render the component with toggle off by default 1`] = `
<body>
  <div>
    <div
      class="compare-ammo-toggle svelte-p12geb"
    >
      <div
        class="
    mdc-form-field
    
    
  "
      >
        <div
          class="mdc-switch"
        >
          <div
            class="mdc-switch__track"
          />
           
          <div
            class="mdc-switch__thumb-underlay"
          >
            <div
              class="mdc-switch__thumb"
            >
              <input
                aria-checked="false"
                class="mdc-switch__native-control "
                id="SMUI-form-field-0"
                role="switch"
                type="checkbox"
              />
            </div>
          </div>
        </div>
         
         
        <label
          for="SMUI-form-field-0"
        >
          <span
            class="ammo-toggle-label svelte-p12geb"
            slot="label"
          >
            Compare
          </span>
        </label>
      </div>
    </div>
  </div>
</body>
`;
